Library/Homebrew/tap.rb

```diff
@@ -280,4 +305,70 @@ def self.each
   def self.names
     map(&:name)
   end
+
+  private
+
+  # @private
+  def formula_file_to_name(file)
+    "#{name}/#{file.basename(".rb")}"
+  end
+
+  # @private
+  def alias_file_to_name(file)
+    "#{name}/#{file.basename}"
+  end
+end
+
+class CoreTap < Tap
+  # @private
+  def initialize
+    @user = "Homebrew"
+    @repo = "homebrew"
+    @name = "Homebrew/homebrew"
+    @path = HOMEBREW_REPOSITORY
+  end
+
+  include Singleton
```

Comment:
I don't see a reason why there needs to be only a single instance here considering there's no local state. It simplifies the code by adding more global state which isn't great for testing in future. Why not just create a single instance rather than using a specific singleton?